### PR TITLE
Duplicated links converted into references in markdown

### LINKED ISSUE #241: Links should be references in markdown
Rather than duplicate the same link many times in markdown, it would be much better to use a reference.

Here's an example:

Instead of 

## Original markdown

```markdown
* Something something [a link](https://mozilla.org/)
* Something else [a link](https://mozilla.org/)
```

We should use:

## Reference markdown

```markdown
* Something something [a link][duplicate link]
* Something else [a link][duplicate link]

[duplicate link]: https://mozilla.org/
```

Both render the same thing, but the reference solution makes updating that link much easier since there's only one!

## Original
* Something something [a link](https://mozilla.org/)
* Something else [a link](https://mozilla.org/)

## Reference version

* Something something [a link][duplicate link]
* Something else [a link][duplicate link]

[duplicate link]: https://mozilla.org/

### PR DESCRIPTION
**Pull request by Patil2099**
Fixes #241

### CODE REVIEW

**Comment by Patil2099** [comment]
@muffinresearch Can you please review this pr ?

**Review by championshuttler — CHANGES_REQUESTED** [review]
Thanks @Patil2099 for the contribution, I think its better to list all the links at the end of the file instead putting them in between the text to keep the file clean and also a standard for markdown files i guess

**Comment by muffinresearch** [comment]
@Patil2099 thanks for this PR I'm going to take a look at this today.

File: src/content/documentation/develop/build-an-accessible-extension.md [hunk 1]
```diff
@@ -90,7 +90,7 @@ Add a shortcut to the button’s action, using the special shortcut `option _exe
 </td>
 <td>
 
-The markup in the popup should follow [standard web accessibility guidelines](https://developer.mozilla.org/docs/Web/Accessibility).
+The markup in the popup should follow [standard web accessibility guidelines].
```
Comment (muffinresearch) [inline src/content/documentation/develop/build-an-accessible-extension.md:93]: This one is missing the same text as the others so it's not converted.

File: src/content/documentation/develop/comparison-with-the-add-on-sdk.md [hunk 2]
```diff
@@ -148,7 +150,7 @@ In both technologies, there are two ways to attach scripts: you can automaticall
 
 | Operation                                    | Add-on SDK                                                                                            | WebExtensions                                                                                                           |
 | -------------------------------------------- | ----------------------------------------------------------------------------------------------------- | ----------------------------------------------------------------------------------------------------------------------- |
-| Attach scripts to pages matching URL pattern | [`page-mod`](https://developer.mozilla.org/docs/Mozilla/Add-ons/SDK/High-Level_APIs/page-mod) API     | [`content-scripts`](https://developer.mozilla.org/docs/Mozilla/Add-ons/WebExtensions/manifest.json/content_scripts) key |
+| Attach scripts to pages matching URL pattern | [`page-mod`][page-mod-link]API     | [`content-scripts`][content-script-link] key |
```
Comment (muffinresearch) [inline src/content/documentation/develop/comparison-with-the-add-on-sdk.md:153]: A space was erroneously deleted before the word "API"

File: src/content/documentation/develop/comparison-with-xul-xpcom-extensions.md [hunk 3]
```diff
@@ -146,11 +146,11 @@ You'll see that many APIs have no WebExtensions equivalent yet. However, we are
...
-| [`nsIClipboard`](https://developer.mozilla.org/docs/Mozilla/Tech/XPCOM/Reference/Interface/nsIClipboard)                                                                                                                                                                                                                                         | Partial: see the [`clipboard`](https://developer.mozilla.org/docs/Mozilla/Add-ons/WebExtensions/API/clipboard) API, and [Interacting with the clipboard](https://developer.mozilla.org/Add-ons/WebExtensions/Interact_with_the_clipboard). |
+| [`nsIClipboard`](https://developer.mozilla.org/docs/Mozilla/Tech/XPCOM/Reference/Interface/nsIClipboard)                                                                                                                                                                                                                                         | Partial: see the [`clipboard`](https://developer.mozilla.org/docs/Mozilla/Add-ons/WebExtensions/API/clipboard) API, and [Interacting with the clipboard](f). |
```
Comment (muffinresearch) [inline src/content/documentation/develop/comparison-with-xul-xpcom-extensions.md:149]: This is not right it's resulting in the link text of 'f'

File: src/content/documentation/develop/differences-between-desktop-and-android-extensions.md [hunk 4]
```diff
@@ -172,3 +172,9 @@ At the time of writing there was an issue with [`storage.sync()`](https://develo
 %}
 
 <!-- END: Single Column Body Module -->
+
+[browser-action]:https://developer.mozilla.org/docs/Mozilla/Add-ons/WebExtensions/API/browserAction
+[browser-action-link]:https://developer.mozilla.org/docs/Mozilla/Add-ons/WebExtensions/manifest.json/browser_action
+[page-action-link]:https://developer.mozilla.org/docs/Mozilla/Add-ons/WebExtensions/manifest.json/page_action
+[page-action-hide]:https://developer.mozilla.org/docs/Mozilla/Add-ons/WebExtensions/API/pageAction/hide
+[page-action]:https://developer.mozilla.org/docs/Mozilla/Add-ons/WebExtensions/API/pageAction/hide
```
Comment (muffinresearch) [inline src/content/documentation/develop/differences-between-desktop-and-android-extensions.md:180]: This is the wrong link. It's the same as the one above.

File: src/content/documentation/develop/request-the-right-permissions.md [hunk 5]
```diff
@@ -209,7 +209,7 @@ A typical scenario for using this approach is the `"geoLocation"` permission. Sa
 
 ![Example of an additional or runtime permission request message](/assets/img/documentation/develop/Permissions_optional_request.png)
 
-Making a runtime permission request is straightforward. Include any permissions you want to request under the `manifest.json` `"[optional_permissions](https://developer.mozilla.org/Add-ons/WebExtensions/manifest.json/optional_permissions)"` key. Then pass the permissions you want granted to [`WebExtAPIRef("permissions.request")`](https://developer.mozilla.org/docs/Mozilla/Add-ons/WebExtensions/API/permissions/request), which prompts the user to grant the permissions. `true` is returned if the user grants the request, `false` if they don't.
+Making a runtime permission request is straightforward. Include any permissions you want to request under the `manifest.json` `"[optional_permissions][optional-permission]"` key. Then pass the permissions you want granted to [`WebExtAPIRef("permissions.request")`](https://developer.mozilla.org/docs/Mozilla/Add-ons/WebExtensions/API/permissions/request), which prompts the user to grant the permissions. `true` is returned if the user grants the request, `false` if they don't.
```
Comment (muffinresearch) [inline src/content/documentation/develop/request-the-right-permissions.md:212]: This was broken before see: https://github.com/mozilla/extension-workshop/issues/926 but back ticks should be inside the square brackets. WIth backticks, the double quotes are redundant.

File: src/content/documentation/develop/test-permission-requests.md [hunk 6]
```diff
@@ -163,4 +163,7 @@ You can then rerun the extension and any runtime permission requests will be dis
 
 <!-- END: Single Column Body Module -->
 
-
+[browser-specific-settings]:https://developer.mozilla.org/docs/Mozilla/Add-ons/WebExtensions/manifest.json/browser_specific_settings
+[nightly]:https://nightly.mozilla.org/
+[developer-link]:https://www.mozilla.org/firefox/developer/
+[upadte-link]:https://developer.mozilla.org/docs/Mozilla/Add-ons/Updates
```
Comment (muffinresearch) [inline src/content/documentation/develop/test-permission-requests.md:169]: This is a typo resulting in a failed replacement.

File: src/content/documentation/publish/make-money-from-browser-extensions.md [hunk 7]
```diff
@@ -259,4 +258,9 @@ Building a large, loyal user base is key to maximizing the revenue you generate,
 
 <!-- END: Single Column Body Module -->
 
-
+[libepay]:https://liberapay.com/
+[micropayment]:https://www.micropayment.de/
+[opencollective]:https://www.micropayment.de/
```
Comment (muffinresearch) [inline src/content/documentation/publish/make-money-from-browser-extensions.md:263]: This is the wrong link for the ref.

**Review by muffinresearch — CHANGES_REQUESTED** [review]
There's a few mistakes in here that are causing broken links that'll need fixing up.

## Improving the references

In addition, here's a few suggestions to improve on the references. 

A good reference needs to be descriptive. Ideally try and avoid using a reference that has a `-link` suffix. The reference should be reasonably clear.

There's a lot places where the linked text could be the reference. This will work if the text that's linked is descriptive (and works from the duplication perspective) you can use just that and no reference. (As mentioned in the [original bug ](https://github.com/mozilla/extension-workshop/issues/241) there's a few ways to do this see [the markdown cheatsheet ](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#links)for the list)

```markdown
Or leave it empty and use the [link text itself].

[link text itself]: http://www.reddit.com
```
 
Which renders as:

Or leave it empty and use the [link text itself].

[link text itself]: http://www.reddit.com

## Testing

To check this I'm diffing the output of a production build from `yarn build:production` on master and from this branch (running the diff on each dist directory - copied each one to a different location to do this). If you can do the same you'll catch any differences that are more than just metadata related.

Overall there should be no changes to the links in the output if this worked correctly.

**Comment by caitmuenster** [comment]
Hey @Patil2099, are you still interested in this issue?

**Comment by Patil2099** [comment]
I will have a look at this on weekend. @caitmuenster

**Comment by Patil2099** [comment]
Sorry, I was busy with some work. I will open a new pr by tomorrow.

**Pull request closed without merging**